homeassistant/components/light/tplink.py

```diff
@@ -119,30 +125,51 @@ def rgb_color(self):
 
     @property
     def is_on(self):
-        """True if device is on."""
+        """Return True if device is on."""
         return self._state
 
     def update(self):
         """Update the TP-Link Bulb's state."""
         from pyHS100 import SmartDeviceException
         try:
+            if self._supported_features == 0:
+                self.get_features()
             self._state = (
                 self.smartbulb.state == self.smartbulb.BULB_STATE_ON)
-            self._brightness = brightness_from_percentage(
-                self.smartbulb.brightness)
-            if self.smartbulb.is_color:
+            if self._name is None:
+                self._name = self.smartbulb.alias
+            if self.smartbulb.is_dimmable:
+                self._brightness = brightness_from_percentage(
+                    self.smartbulb.brightness)
+            if self.smartbulb.is_variable_color_temp:
                 if (self.smartbulb.color_temp is not None and
                         self.smartbulb.color_temp != 0):
                     self._color_temp = kelvin_to_mired(
                         self.smartbulb.color_temp)
+            if self.smartbulb.is_color:
                 self._rgb = hsv_to_rgb(self.smartbulb.hsv)
+            if self.smartbulb.has_emeter:
+                self._emeter_params[ATTR_CURRENT_CONSUMPTION] \
+                    = "%.1f W" % self.smartbulb.current_consumption()
+                daily_statistics = self.smartbulb.get_emeter_daily()
+                monthly_statistics = self.smartbulb.get_emeter_monthly()
+                self._emeter_params[ATTR_DAILY_CONSUMPTION] \
+                    = "%.2f kW" % daily_statistics[int(time.strftime("%d"))]
```

Comment:
Looks like `get_emeter_daily()` and `get_emeter_monthly()` will return and empty list when there is no information available on that time point, which may cause this to crash (especially if the bulb has not have an access to external network to synchronize its time, like is the case for those who are just using this over the wlan).

I think this should at the end be fixed at pyhs100's end, but for now you will want to do as done in the switch platform and wrap it with a try-catch block.